src/python/numpy_helper.hpp

```diff
@@ -160,7 +165,12 @@ struct build_array_from_range_impl<Iterator, typename std::enable_if<std::is_sam
       size_t i = 0;
       for (Iterator it = range.first; it != range.second; it++, i++) {
         PyObject **ref = (PyObject **)PyArray_GETPTR1((PyArrayObject*)array, i);
-        PyObject *newobj = PyString_FromStringAndSize((*it).c_str(), (*it).size());
+        PyObject *newobj = NULL;
+        #if PY_MAJOR_VERSION < 3
+        newobj = PyString_FromStringAndSize((*it).c_str(), (*it).size());
+        #else
+        newobj = PyUnicode_FromStringAndSize((*it).c_str(), (*it).size());
+        #endif
```

Comment:
Put this in an inline helper method?

`PyUnicode_FromStringAndSize` assumes UTF-8 -- I'm creating an issue about other string encodings (or you can force bytes and decode after parsing)